### ISSUE TITLE
Ubuntu16.04: Menu is missing

### ISSUE THREAD

**Issue by stdedos** [keepassxreboot/keepassxc#3066]
[TIP]:  # ( Provide a general summary of the issue in the title above ^^ )
[TIP]:  # ( DO NOT include screenshots of your actual database! )

_Issue might be needed to be split later. However, findings were discovered together_

## Expected Behavior
[NOTE]: # ( Tell us what you expected to happen )
* `KeePass 2.3.4+`: Being able to use the menus
* `KeePass 2.4.1`: No crashes

## Current Behavior
[NOTE]: # ( Tell us what actually happens )
* `KeePass 2.3.4+`: Menus don't render.
* `KeePass 2.4.1`: App crashes inexplicably and invisibly once in a while.

## Possible Solution
[NOTE]: # ( Not required, but suggest a fix/reason for the bug )
For some reason, I am missing the top menus in Ubuntu 16.04.
Preliminary debuggins shows that, if
```
ShowTrayIcon=true
SplitterState=119, 924
ToolButtonStyle=0 
```
are set (as opposed to an "empty" `ini` file), then the menus are not shown.

## Steps to Reproduce
[NOTE]: # ( Provide a link to a live example, or an unambiguous set of steps to )
[NOTE]: # ( reproduce this bug. Include code to reproduce, if relevant )
1.
2.
3.

## Context
[NOTE]: # ( How has this issue affected you? What unique circumstances do you have? )

## Debug Info
[NOTE]: # ( Paste debug info from Help → About here )
KeePassXC - Version 2.4.1
Revision: 7bafe65

Qt 5.5.1
Debugging mode is disabled.

Operating system: Ubuntu 16.04.6 LTS
CPU architecture: x86_64
Kernel: linux 4.15.0-45-generic

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (signed and unsigned sharing)
- YubiKey

Cryptographic libraries:
 libgcrypt 1.8.1

**Comment by stdedos** [comment]
It could be that 
```
ToolButtonStyle=0 
```
could be responsible for 
>  For some reason, I am missing the top menus in Ubuntu 16.04.

and 

```
ShowTrayIcon=False
```
for the crash.
> KeePass 2.4.1: Application also crashes at startup inexplicably and invisibly once in a while.

**Comment by droidmonkey** [comment]
Why are you manually editing the ini file. We have a settings gui for a reason.

**Comment by stdedos** [comment]
> Why are you manually editing the ini file. We have a settings gui for a reason.

I am not. I simply tried to debug the issue, without losing my current settings.
"All of those" were part of my original `keepassxc.ini`.
I removed that file, and incrementally re-merged it to bisect.

**Comment by droidmonkey** [comment]
Gotcha

**Comment by phoerious** [comment]
The crash could be the known QTreeView bug in Qt 5.5, which is totally unrelated (try the AppImage, which has a newer Qt version and doesn't crash).0

The menu bar issue could be a reincarnation of #271, although this should be fixed by our workaround.

**Comment by stdedos** [comment]
> The crash could be the known QTreeView bug in Qt 5.5, which is totally unrelated (try the AppImage, which has a newer Qt version and doesn't crash). We need a stack trace to be sure.

I can collect one, if the steps are known.

Can I in-place use the AppImage?
Since I'd like auto updates, would `snap` package be a suiteable alternative?

> The menu bar issue could be a reincarnation of #271, although this should be fixed by our workaround.

You mean the Qt-update workaround?

**Comment by phoerious** [comment]
You need a debug build and run it in GDB to create a usable stack trace. Build instructions are here: https://github.com/keepassxreboot/keepassxc/wiki/Building-KeePassXC

The AppImage can do auto updates via appimaged, but you can also use the Snap if you prefer something that integrates better into your system.

> You mean the Qt-update workaround?

We send the missing dbus signal ourselves if the Qt version KeePassXC was compiled with is too old. The bug caused menus to vanish if KeePassXC is hidden to the tray and not be restored afterwards. It was fixed in Qt 5.9 or so (16.04 has Qt 5.5).

**Comment by stdedos** [comment]
After some fiddling (including an `apt-get purge`), it seems the "crash" could be `MinimizeOnStartup=true` (I saw the dash icon coming on and off, but I didn't notice the tray icon opening). I disabled it, so that I don't make the same mistake again.

However, now with my original `ini` file (minus the above setting), I don't see neither issue. So, either `apt-get purge` fixed it, or some weird interaction with `MinimizeOnStartup=true`. (I also purged `snap` and `AppImage`)

I won't generate the stacktrace, as it seems that this could be my issue/configuration issue. Feel free to handle it as you wish.
